agent/src/main/java/com/teamscale/jacoco/agent/commit_resolution/git_properties/GitPropertiesLocatorUtils.java

```diff
@@ -147,25 +175,23 @@ public static ProjectRevision getProjectRevisionFromGitProperties(
 		}
 	}
 
-	private static Pair<String, Properties> findGitPropertiesInDirectoryFile(File file) throws IOException {
-		File[] directoryFiles = file.listFiles();
-		if (directoryFiles == null) {
-			return null;
+	private static Pair<String, Properties> findGitPropertiesInDirectoryFile(File directoryFile) throws IOException {
+		List<File> directoryFiles = FileSystemUtils.listFilesRecursively(directoryFile,
+				file -> file.getName().toLowerCase().equals(GIT_PROPERTIES_FILE_NAME));
+		if (directoryFiles.size() != 1) {
+			throw new IllegalArgumentException(
+					"There must be a single git.properties file in the directory " + directoryFile.toString());
```

Comment:
Would help in debugging if we would include the paths under which we found git.properties as the files could get quite big.

Must there be a git.properties in every scanned folder path? Or could there be multiple with only a single git.properties?

Reply:
Ah, you're right. I forgot it might simply be a jar/war with no git.properties inside. Then we just scan the next.

However, I would still expect a single git.properties file to be present in a single jar/war, if any. Otherwise, we wouldn't know which values to adopt, which is a problem.

Reply:
I don't quite get the comment regarding debugging. Do you really want to me log the exact locations in case we find multiple git.properties files in a single jar? This is both 
 vary unlikely and can get quite lengthy.

Reply:
Yes, that is what I meant. In case you really run into such a problem I would rather have too much information then not knowing what is going on. In case you really end up with two git.properties you most likely did not expect this to happen and it makes it easier to spot the error if you see the two paths side by side that the agent found.

Reply:
Done.